backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/utils/Shapes.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.utils
+
+import androidx.compose.foundation.shape.GenericShape
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.unit.LayoutDirection
+import androidx.compose.ui.util.lerp
+
+internal fun DeflatedRect(
```

Comment:
And this is an implementation of easy way to create a semi-filled rects

Reply:
The name is inspired by `Rect.deflate()` method in Compose. It draws only a part of the rect – specified by its range parameters. vertical = 0.5f..1f will only draw the bottom half of the rect.